### DISCUSSION TITLE
lazyLoading on MultiSelect (15.4.13) always generation event.first = 0

### DISCUSSION THREAD

**Discussion by ppaulis** [primefaces/community#623] [PrimeNG]
Hi all,

I'm currently trying to implement lazyLoading with virtualScroll in a multiSelect because the input data becomes too heavy. It seems that the generated event for `onLazyLoad` always looks like this : `{first: 0, last: ...}`, meaning, `first` is always 0. And because of this, I guess, the multiSelect always jumps back to entry 0 when scrolling down.

My multiSelect is configured like this :
```
<p-multiSelect
      inputId="customers"
      [options]="customers"
      [(ngModel)]="selectedCustomers"
      showToggleAll="false"
      optionLabel="name"
      filter="true"
      placeholder="Select Customer(s)"
      (onPanelHide)="filterUpdated()"
      showClear="true"
      [virtualScroll]="true"
      [virtualScrollItemSize]="30"
      lazy="true"
      (onLazyLoad)="lazyLoadCustomers($event)"
```

where `lazyLoadCustomers()` is more or less something like this :
```
lazyLoadCustomers(event) {
  this.customerService.loadCustomers(event.first, event.last - event.first + 1).subscribe(result => {
    this.customers = result;
  });
}
```
The 2 parameters for `loadCustomers()` simply are offset and result size.

I guess that I'm misunderstanding some concepts here.. So if someone could point me to an example or documentation that I missed, that would be much appreciated!

Thanks!

**Comment by cetincakiroglu** [comment]
Hi,

Could you please share a stackblitz example so we can inspect and understand the issue clearly?

**Comment by cetincakiroglu** [comment]
Closing the topic due to the inactivity.
